gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/importing/ImportBundleTask.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayexport.tasks.importing;
+
+import com.ca.apim.gateway.cagatewayexport.config.GatewayConnectionProperties;
+import com.ca.apim.gateway.cagatewayexport.util.http.GatewayClient;
+import com.ca.apim.gateway.cagatewayexport.util.http.GatewayClientException;
+import org.apache.http.entity.StringEntity;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.File;
+import java.io.IOException;
+
+import static com.ca.apim.gateway.cagatewayexport.util.injection.ExportPluginModule.getInstance;
+import static java.nio.charset.Charset.defaultCharset;
+import static org.apache.commons.io.FileUtils.readFileToString;
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+import static org.apache.http.client.methods.HttpPut.METHOD_NAME;
+import static org.apache.http.client.methods.RequestBuilder.create;
+
+/**
+ * Task to import a bundle into an existing running gateway.
+ */
+public class ImportBundleTask extends DefaultTask {
+
+    private GatewayClient gatewayClient;
+
+    //Inputs
+    private GatewayConnectionProperties gatewayConnectionProperties;
+    private RegularFileProperty importFile;
+
+    public ImportBundleTask() {
+        this.gatewayClient = getInstance(GatewayClient.class);
+        gatewayConnectionProperties = new GatewayConnectionProperties(getProject());
+        importFile = newInputFile();
+
+        // makes it so that the export is always run
+        getOutputs().upToDateWhen(t -> false);
+    }
+
+    @Nested
+    public GatewayConnectionProperties getGatewayConnectionProperties() {
+        return gatewayConnectionProperties;
+    }
+
+    public void setGatewayConnectionProperties(GatewayConnectionProperties gatewayConnectionProperties) {
+        this.gatewayConnectionProperties = gatewayConnectionProperties;
+    }
+
+    /**
+     * @return the bundle file to be imported
+     */
+    public RegularFileProperty getImportFile() {
+        return importFile;
+    }
+
+    @TaskAction
+    public void perform() {
+        File bundleFile = importFile.getAsFile().get();
```

Comment:
How do you specify which file gets imported?

Reply:
Something like this in your gradle file:
```
project.afterEvaluate {
    tasks.getByName("import-bundle").importFile = tasks.getByName("build-full-bundle").outputBundle
}
```

I will add to documentation later.

Reply:
I see, this works. We can consider making a GatewayImportConfig or something like that in the future in case there are multiple things we want to upload to a running Gateway

Reply:
for now I am going to leave this way and we get some feedback on the usage, if gets too much complicated we change :)

Reply:
Also, we need to think of future roadmap where no one modifies running gateway, this is something for intermediate state until we reach that perfect ephemeral state.